### PR TITLE
for the method Html::ul(); added work with multidimensional arrays.

### PR DESCRIPTION
**Pull request by den67rus**
add: Added documentation in Russian

### CODE REVIEW

File: docs/guide-ru/helper-html.md [hunk 1]
```diff
@@ -429,16 +429,99 @@ echo Html::getAttributeName('dates[0]');
...
+                '2.2.1 item',
+            ],
+        ],
+        '3 item',
+    ];
+    
+    // Массив с опциями
+    $options = [
+        'children' => true,     // Включает поддрежку многомерных массивов (по умолчанию выключено)
+        'encode' => false,      // Отключаем HTMl экранинование (по умолчанию включено)
```
Comment (samdark) [inline docs/guide-ru/helper-html.md]: HTML

File: docs/guide-ru/helper-html.md [hunk 2]
```diff
@@ -429,16 +429,99 @@ echo Html::getAttributeName('dates[0]');
...
+        'class' => 'ulclass',   // Установим класс для тега <ul>
```
Comment (samdark) [inline docs/guide-ru/helper-html.md]: What about nested UL tags?

File: docs/guide-ru/helper-html.md [hunk 3]
```diff
@@ -429,16 +429,99 @@ echo Html::getAttributeName('dates[0]');
...
 
-Для формирования упорядоченных списков используйте `Html::ol()`.
\ No newline at end of file
+Каждый элемент массива генерирует один элемент списка. Для включения многомерных массивов используется опция `children`, по умолчанию она отключена.
+При генерации многомерных массивов дочерний элемент списка вкладывается в идущий выше родительский. Если передать 
+пустую переменную $items то сгенерируются пустые теги <ul></ul> или <ol></ol> в зависимости от вызываемого метода. 
+Для каждого элемента списка с помощью `itemOptions` можно установить опции по такому же принципу, как и для метода `Html::tag()`.
+Обратите внимание, что содержимое будет автоматически экранироваться c помощью `Html::encode` если не задействована опция `$options['encode'] = false;`.
+
+В результате на выходе от вышеприведённого примера получаем следующий html код
```
Comment (samdark) [inline docs/guide-ru/helper-html.md]: HTML

File: framework/helpers/BaseHtml.php [hunk 4]
```diff
@@ -979,7 +979,8 @@ public static function radioList($name, $selection = null, $items = [], $options
     /**
      * Generates an unordered list.
      * @param array|\Traversable $items the items for generating the list. Each item generates a single list item.
-     * Note that items will be automatically HTML encoded if `$options['encode']` is not set or true.
+     * Note that items will be automatically HTML encoded if `$options['encode']` is not set or true. The same way
```
Comment (samdark) [inline framework/helpers/BaseHtml.php]: No need to mention multidimension array here.

File: framework/helpers/BaseHtml.php [hunk 5]
```diff
@@ -995,6 +996,24 @@ public static function radioList($name, $selection = null, $items = [], $options
      *   where $index is the array key corresponding to `$item` in `$items`. The callback should return
      *   the whole list item tag.
      *
+     * - children: boolean, support multidimensional array. Defaults to false.
```
Comment (samdark) [inline framework/helpers/BaseHtml.php]: It's usually called `items`.

File: framework/helpers/BaseHtml.php [hunk 6]
```diff
@@ -1003,29 +1022,50 @@ public static function ul($items, $options = [])
     {
         $tag = isset($options['tag']) ? $options['tag'] : 'ul';
         $encode = !isset($options['encode']) || $options['encode'];
+        $children = isset($options['children']) && $options['children'] === true;
```
Comment (samdark) [inline framework/helpers/BaseHtml.php]: I'm not sure option is really useful.

File: framework/helpers/BaseHtml.php [hunk 7]
```diff
@@ -1003,29 +1022,50 @@ public static function ul($items, $options = [])
...
         }
+        $results[] = static::endTag('li');
         return static::tag($tag, "\n" . implode("\n", $results) . "\n", $options);
     }
 
     /**
      * Generates an ordered list.
      * @param array|\Traversable $items the items for generating the list. Each item generates a single list item.
-     * Note that items will be automatically HTML encoded if `$options['encode']` is not set or true.
+     * Note that items will be automatically HTML encoded if `$options['encode']` is not set or true. The same way
```
Comment (samdark) [inline framework/helpers/BaseHtml.php]: No need to mention multidimensional array.

**Comment by samdark** [comment]
1. Need docs in English as well.
2. Is it for menus? Have you checked `yii\widgets\Menu`?

**Comment by cebe** [comment]
@den67rus are you willing to work on this further?

**Comment by cebe** [comment]
For our issue tracker to be effective I am closing this. If you want to work on it, please let us know and we will reopen it.

**Pull request closed without merging**